ui/libs/__init__.py

```diff
@@ -69,6 +97,14 @@ def clone_user_data(extensions_dir: pathlib.Path) -> pathlib.Path:
     )
 
 
+def extract_tar_gz(source: pathlib.Path, dest: pathlib.Path) -> pathlib.Path:
+    """Extract source into destination"""
+    file = tarfile.open(source)
```

Comment:
tarfile support work use "with" operator, please use it

Reply:
Good idea! I added `with` statement!